File: server/backend/app/models/item.py
from sqlalchemy import Column, Integer, String, ForeignKey, DateTime
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from ..database import Base

class Item(Base):
    __tablename__ = "item"

    item_id = Column(Integer, primary_key=True, index=True)
    name = Column(String(255), nullable=False)
    location = Column(String(255))
    quantity = Column(Integer, default=1, nullable=False)
    warehouse_id = Column(Integer, ForeignKey("warehouse.warehouse_id"), nullable=False)
    category_id = Column(Integer, ForeignKey("category.category_id"), nullable=True) # New category_id field
    created_at = Column(DateTime, server_default=func.now())
    updated_at = Column(DateTime, server_default=func.now(), onupdate=func.now())
    deleted_at = Column(DateTime, nullable=True) # Soft delete column

    warehouse = relationship("Warehouse", back_populates="items")
    category = relationship("Category", back_populates="items") # New relationship
    media = relationship("ItemMedia", back_populates="item")

File: app/lib/models/item.dart
import 'item_media.dart';
import 'category.dart'; // Import Category model

class Item {
  final int id;
  final String name;
  final Category? category;
  final String? location;
  final int quantity;
  final int warehouseId;
  final List<ItemMedia> media;
  final String? deletedAt;

  Item({
    required this.id,
    required this.name,
    this.category,
    this.location,
    required this.quantity,
    required this.warehouseId,
    this.media = const [],
    this.deletedAt,
  });

  factory Item.fromJson(Map<String, dynamic> json) {
    var mediaList = json['media'] as List? ?? [];
    List<ItemMedia> media = mediaList.map((i) => ItemMedia.fromJson(i)).toList();

    return Item(
      id: json['item_id'],
      name: json['name'],
      category: json['category'] != null ? Category.fromJson(json['category']) : null,
      location: json['location'],
      quantity: json['quantity'],
      warehouseId: json['warehouse_id'],
      media: media,
      deletedAt: json['deleted_at'],
    );
  }
}
